app/controllers/media/objects_controller.rb

```diff
@@ -5,9 +5,15 @@ class Media::ObjectsController < ApplicationController
   def show
     style = params[:style]
     @answer = @media_object.answer
-    @response = @answer.response
+    @response = @answer.try(:response)
     disposition = (params[:dl] == "1") ? "attachment" : "inline"
-    authorize! :show, @response
+
+    if @response
+      authorize! :show, @response
+    else
+      authorized = @media_object.token == params[:token]
+      raise CanCan::AccessDenied.new("Not authorized to show media", :view, :media_object) unless authorized
+    end
```

Comment:
The explicit skip is already at the top of the controller because we aren't using the auto-authorizers since we're auth-ing the response and didn't write custom logic for media objects.